### PR TITLE
Handle bigint and boolean casts in mysql json unwrapping

### LINKED ISSUE #24720: MySQL unfolding of JSON fails for `boolean` and `bigint` attributes
**Describe the bug**
MySQL unfolding of JSON fails for `boolean` and `bigint` attributes - and perhaps other types too.

Workaround is the same as described in https://github.com/metabase/metabase/issues/22732#issuecomment-1127424455, disable JSON unfolding.

**To Reproduce**
1. Create a MySQL 8 table like this:
```
CREATE TABLE `x_jsontest` (`jsoncol` json DEFAULT NULL) ENGINE=InnoDB;
INSERT INTO `x_jsontest` (`jsoncol`) VALUES ('{"mybool": true, "myint": 1234567890123456}');
```
2. Metabase > Admin > Databases > (mysql) > Sync database schema now - there's be errors during fingerprinting
3. New question > (mysql) > (x_jsontest) - cannot view the table, it fails with:
`You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'BIGINT)`... or ...`near 'BOOLEAN)`...

**Expected behavior**
Don't break existing tables if they contain JSON. It's now possible to disable unfolding, but then you'll have to manually set that, do a sync, and then manually go through all tables in Data Model, since the JSON columns has been set to Visibility="Only in detail view".

**Information about your Metabase Installation:**
Tested 0.44.0 with MySQL 8 data source

### PR DESCRIPTION
**Pull request by escherize**
- Add JSON -> SerializedJSON to `database-type->base-type`

Fixes #24720 - 3 issues with mysql json:

1. Fingerprinting json columns was breaking on mysql. It works now
2. bigint cast output of exploded json columns now goes through the same path as a regular field
3. boolean cast output of exploded json columns now goes through the same path as a regular field

### CODE REVIEW

File: src/metabase/driver/mysql.clj [hunk 1]
```diff
@@ -40,10 +40,8 @@
 
 (defmethod driver/display-name :mysql [_] "MySQL")
 
-(defmethod driver/database-supports? [:mysql :nested-field-columns] [_ _ database]
-  (or (get-in database [:details :json-unfolding]) true))
```
[the remark this answers is not in the document]
Comment (flamber) [reply src/metabase/driver/mysql.clj:44]: That's a P1 bug #25068

[the remark this answers is not in the document]
Comment (escherize) [reply src/metabase/driver/mysql.clj:44]: I reverted this change here, handled in: [#25556 - allow turning off mysql json unwrapping](https://github.com/metabase/metabase/pull/25556).

**Review by noahmoss — APPROVED** [review]
Don't know much about mysql types so I'll trust that these fix the issues, but just left a couple minor clojure comments

**Comment by escherize** [comment]
Good points all around -- thank you @noahmoss

**Pull request merged**